core/src/main/java/hudson/util/IOUtils.java

```diff
@@ -51,20 +51,11 @@ public static void copy(InputStream in, File out) throws IOException {
      *      This method returns the 'dir' parameter so that the method call flows better.
      */
     public static File mkdirs(File dir) throws IOException {
-        if(dir.mkdirs() || dir.exists())
-            return dir;
-
-        // following Ant <mkdir> task to avoid possible race condition.
         try {
-            Thread.sleep(10);
-        } catch (InterruptedException e) {
-            // ignore
+            return Files.createDirectories(dir.toPath()).toFile();
+        } catch (UnsupportedOperationException e) {
```

Comment:
Also https://docs.oracle.com/javase/7/docs/api/java/nio/file/InvalidPathException.html . @dwnusbaum has added a utility method for that recently, so you could replace `dir.toPath()` byt `fileToPath(dir)`